docs/adr/0008-managementv3-clusters-support.md

```diff
@@ -0,0 +1,42 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [8. Management V3 clusters support](#8-management-v3-clusters-support)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 8. Management V3 clusters support
+
+* Status: proposed
+* Date: 2024-01-25
+* Authors: @salasberryfin
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+As an operator, I'd like to have an alternative import strategy that will generate a `management.cattle.io/v3` cluster instead of a `provisioning.cattle.io/v1` cluster. This is to ensure that we can support this scenario in light of future changes to Rancher.
+
+## Decision
+
+The current import controller logic creates a `provisioning.cattle.io/v1` cluster when a new CAPI cluster is detected. To support a new import strategy, based on a new feature flag, Rancher Turtles will either start the existing `provisioning.cattle.io` controller or a new one for `management.cattle.io`.
+
+Adding support for this resource presents some challenges:
+- Owner Reference cannot be set: `management.cattle.io/v3` clusters are cluster-scoped while CAPI clusters are namespaced.
+- Display name and resource name are different: the resource name must satisfy the regex `^(c-[a-z0-9]{5}|local)$`.
```

Comment:
If a conflict occurs, name can be randomly generated again until the cluster is applied. Using only hash does not allow this.

Reply:
If we randomly generate the name, we know that a conflict occurs when creating the cluster resource [here](https://github.com/rancher-sandbox/rancher-turtles/blob/8109fda7fe2d77f549ea03fcb0c9b2095e5dd55a/internal/controllers/import_controller.go#L234), but then, how can we actually fetch the correct cluster resource [here](https://github.com/rancher-sandbox/rancher-turtles/blob/8109fda7fe2d77f549ea03fcb0c9b2095e5dd55a/internal/controllers/import_controller.go#L199) if that was a random string?

Reply:
When an owner reference is set on the CAPI cluster, we can track it from there. Otherwise the cluster v3 needs to be labeled with the CAPI cluster name to allow lookup.

Reply:
When we create the `v3.Cluster` we can just use the builtin functionality of Kubernetes to ensure the resource has a unique name. Using `GenerateName: c-` and not setting `Name` when we create will be enough.